types/lodash/lodash-tests.ts

```diff
@@ -11556,65 +11606,274 @@ namespace TestMapKeys {
     let array: AbcObject[] | null | undefined = [] as any;
     let list: _.List<AbcObject>| null | undefined = [] as any;
     let dictionary: _.Dictionary<AbcObject> | null | undefined = anything;
+    let numericDictionary: _.NumericDictionary<AbcObject> | null | undefined = anything;
+    let abcObject: AbcObject = anything;
 
     let listIterator = (value: AbcObject, index: number, collection: _.List<AbcObject>) => "";
     let dictionaryIterator = (value: AbcObject, key: string, collection: _.Dictionary<AbcObject>) => "";
+    let numericDictionaryIterator = (value: AbcObject, key: string, collection: _.NumericDictionary<AbcObject>) => "";
+    let abcObjectIterator = (value: AbcObject[keyof AbcObject], key: string, collection: AbcObject) => "";
 
     {
-        let result: _.Dictionary<AbcObject>;
+        _.mapKeys(array); // $ExpectType Dictionary<AbcObject>
+        _.mapKeys(array, listIterator); // $ExpectType Dictionary<AbcObject>
+        _.mapKeys(array, ''); // $ExpectType Dictionary<AbcObject>
+        _.mapKeys(array, {}); // $ExpectType Dictionary<AbcObject>
 
-        result = _.mapKeys(array);
-        result = _.mapKeys(array, listIterator);
-        result = _.mapKeys(array, '');
-        result = _.mapKeys(array, {});
+        _.mapKeys(list); // $ExpectType Dictionary<AbcObject>
+        _.mapKeys(list, listIterator); // $ExpectType Dictionary<AbcObject>
+        _.mapKeys(list, ''); // $ExpectType Dictionary<AbcObject>
+        _.mapKeys(list, {}); // $ExpectType Dictionary<AbcObject>
 
-        result = _.mapKeys(list);
-        result = _.mapKeys(list, listIterator);
-        result = _.mapKeys(list, '');
-        result = _.mapKeys(list, {});
+        _.mapKeys(dictionary); // $ExpectType Dictionary<AbcObject>
+        _.mapKeys(dictionary, dictionaryIterator); // $ExpectType Dictionary<AbcObject>
+        _.mapKeys(dictionary, ''); // $ExpectType Dictionary<AbcObject>
+        _.mapKeys(dictionary, {}); // $ExpectType Dictionary<AbcObject>
 
-        result = _.mapKeys(dictionary);
-        result = _.mapKeys(dictionary, dictionaryIterator);
-        result = _.mapKeys(dictionary, '');
-        result = _.mapKeys(dictionary, {});
+        _.mapKeys(numericDictionary); // $ExpectType Dictionary<AbcObject>
+        _.mapKeys(numericDictionary, numericDictionaryIterator); // $ExpectType Dictionary<AbcObject>
+        _.mapKeys(numericDictionary, ''); // $ExpectType Dictionary<AbcObject>
+        _.mapKeys(numericDictionary, {}); // $ExpectType Dictionary<AbcObject>
```

Comment:
These tests don't seem to work in TS 2.4 - it gets a return type of `Dictionary<any>`. I guess it's best to just remove these tests.